mesos-client/src/main/scala/mesosphere/mesos/client/MesosCallFactory.scala

```diff
@@ -0,0 +1,235 @@
+package mesosphere.mesos.client
+
+import akka.util.ByteString
+import com.google.protobuf
+import org.apache.mesos.v1.mesos.{ AgentID, ExecutorID, FrameworkID, Filters, KillPolicy, OfferID, Request, TaskID }
+import org.apache.mesos.v1.scheduler.scheduler.Call.{ Accept, Acknowledge, Decline, Kill, Message, Reconcile, Revive }
+import org.apache.mesos.v1.scheduler.scheduler.{ Call, Event }
+
+class MesosCallFactory(frameworkId: FrameworkID) {
+  private val someFrameworkId = Some(frameworkId)
+  /**
+    * ***************************************************************************
+    * Helper methods to create mesos `Call`s
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#calls
+    * ***************************************************************************
+    */
+
+  /**
+    * Factory method to construct a TEARDOWN Mesos Call event. Calling this method has no side effects.
+    *
+    * This event is sent by the scheduler when it wants to tear itself down. When Mesos receives this request it will
+    * shut down all executors (and consequently kill tasks). It then removes the framework and closes all open
+    * connections from this scheduler to the Master.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#teardown
+    */
+  def teardown(): Call = {
+    Call(
+      frameworkId = someFrameworkId,
+      `type` = Some(Call.Type.TEARDOWN)
+    )
+  }
+
+  /**
+    * Factory method to construct a ACCEPT Mesos Call event. Calling this method has no side effects.
+    *
+    * Sent by the scheduler when it accepts offer(s) sent by the master. The ACCEPT request includes the type
+    * of operations (e.g., launch task, launch task group, reserve resources, create volumes) that the scheduler
+    * wants to perform on the offers. Note that until the scheduler replies (accepts or declines) to an offer,
+    * the offer’s resources are considered allocated to the offer’s role and to the framework.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#accept
+    */
+  def accept(accepts: Accept): Call = {
```

Comment:
What about passing the parameters that accept takes as done with `decline`?

Reply:
This is a good point; @zen-dog wdyt? I just moved the method from a previous location.